Models/AbstractProjectModels/Validation/Data/User/Validator.php

```diff
@@ -0,0 +1,153 @@
+<?php
+declare(strict_types=1);
+
+namespace Models\AbstractProjectModels\Validation\Data\User;
+
+use Interfaces\User\UserDataValidatorInterface;
+
+abstract class Validator implements UserDataValidatorInterface
+{
+    private const REGISTRATION_FIELDS = [
+        'login',
+        'pass',
+        'pass_confirm',
+        'name',
+        'birthdate',
+        'email'
+    ];
+    private const LOGIN_FIELDS = [
+        'login',
+        'pass'
+    ];
+    protected array $requiredFields = [];
+
+    public function __construct()
+    {
+        $this->requiredFields = [
```

Comment:
Why can't you specify this array in the class property itself? I don't see a reason to overwrite existing $requiredFields in the constructor while instance is being created. Better to specify them to a property initially